### PR TITLE
Update exports to use openpyxl; export xlsx

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/flows/models.py [hunk 1]
```diff
@@ -3787,70 +3781,61 @@ def as_org_tz(dt):
...
+
+            sheet = book[sheet_name]
             # build up our header row
 
-            index = 0
+            index = 1
             if show_submitted_by:
-                sheet.write(0, index, "Surveyor")
-                sheet.col(0).width = medium_width
+                sheet.cell(row=1, column=index, value="Surveyor")
```
Comment (ericnewcomer) [inline temba/flows/models.py]: Really, they are one based?

File: temba/flows/models.py [hunk 2]
```diff
@@ -3914,36 +3899,31 @@ def get_contact_urn_display(contact):
...
 
                 # a new run
                 if last_run != run_step.run.pk:
-                    earliest = run_step.arrived_on
+                    earliest = run_step.arrived_on.replace(microsecond=0)
                     latest = None
 
                     if include_runs:
-                        if run_row % 1000 == 0:
-                            runs.flush_row_data()
```
Comment (ericnewcomer) [inline temba/flows/models.py]: Is there a corollary here? Are we going to be able to export very large documents without a flush to disk? Is it handled implicitly in the new library?

File: temba/flows/models.py [hunk 3]
```diff
@@ -3988,49 +3968,49 @@ def get_contact_urn_display(contact):
...
                 if include_runs:
-                    runs.write(run_row, padding + 4 + cf_padding, as_org_tz(earliest), date_format)
-                    runs.write(run_row, padding + 5 + cf_padding, as_org_tz(latest), date_format)
+                    runs.cell(row=run_row, column=padding + 4 + cf_padding, value=as_org_tz(earliest))
+                    runs.cell(row=run_row, column=padding + 5 + cf_padding, value=as_org_tz(latest))
 
-                merged_runs.write(merged_row, padding + 4 + cf_padding, as_org_tz(merged_earliest), date_format)
-                merged_runs.write(merged_row, padding + 5 + cf_padding, as_org_tz(merged_latest), date_format)
+                merged_runs.cell(row=merged_row, column=padding + 4 + cf_padding, value=as_org_tz(merged_earliest))
+                merged_runs.cell(row=merged_row, column=padding + 5 + cf_padding, value=as_org_tz(merged_latest))
```
Comment (ericnewcomer) [inline temba/flows/models.py]: Are you sure that dates are formatted properly in the resulting file?

File: temba/flows/models.py [hunk 4]
```diff
@@ -4043,42 +4023,34 @@ def get_contact_urn_display(contact):
...
-                        msgs.write(0, 5, "Message")
-                        msgs.write(0, 6, "Channel")
-
-                        msgs.col(0).width = medium_width
-                        msgs.col(1).width = small_width
-                        msgs.col(2).width = medium_width
-                        msgs.col(3).width = medium_width
-                        msgs.col(4).width = small_width
-                        msgs.col(5).width = large_width
-                        msgs.col(6).width = small_width
```
Comment (ericnewcomer) [inline temba/flows/models.py]: Same here, setting default column widths are kind of nice in that it ensures things look natural upon opening.

File: temba/flows/models.py (hunk 1 above)
Comment (norkans7) [inline temba/flows/models.py]: yep, it even throws when one use 0

File: temba/flows/models.py (hunk 2 above)
Comment (norkans7) [inline temba/flows/models.py]: It seems we need https://openpyxl.readthedocs.io/en/default/optimized.html#write-only-mode instead to keep using low memory, I'll update the PR

File: temba/flows/models.py (hunk 3 above)
Comment (norkans7) [inline temba/flows/models.py]: previous 12/09/2015 12:00:46 
current 2015-12-04 18:51:25

I'll look closely how to adjust the date format

File: temba/flows/models.py (hunk 4 above)
Comment (norkans7) [inline temba/flows/models.py]: I'll look close on this too, there must be a way to set this

File: temba/utils/tests.py [hunk 5]
```diff
@@ -778,29 +785,33 @@ def test_xls(self):
         for i in range(32):
             values.append("Value %d" % i)
 
-        # write out 67,000 rows, that'll make two sheets
-        for i in range(67000):
+        # write out 1050000 rows, that'll make two sheets
```
Comment (ericnewcomer) [inline temba/utils/tests.py]: this comment doesn't look right.

**Comment by norkans7** [comment]
@ericnewcomer I update to use write only mode which should be memory efficient for large exports https://openpyxl.readthedocs.io/en/default/optimized.html#write-only-mode

Please review once again

**Comment by ericnewcomer** [comment]
We'll merge this with imports at the same time.

https://github.com/nyaruka/rapidpro/pull/878

**Pull request closed without merging**